indico/modules/events/registration/fields/sessions.py

```diff
@@ -0,0 +1,56 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2024 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from itertools import groupby
+
+from marshmallow import ValidationError, fields, validate
+
+from indico.modules.events.registration.fields.base import RegistrationFormFieldBase
+from indico.modules.events.sessions.models.blocks import SessionBlock
+from indico.util.i18n import _
+
+
+class SessionsField(RegistrationFormFieldBase):
+    name = 'sessions'
+    mm_field_class = fields.List
+    mm_field_args = (fields.Integer,)
+    setup_schema_fields = {
+        'display': fields.String(load_default='expanded'),
+        'allow_full_days': fields.Boolean(load_default=False),
+        'minimum': fields.Integer(load_default=0, validate=validate.Range(0)),
+        'maximum': fields.Integer(load_default=0, validate=validate.Range(0)),
+    }
+
+    @property
+    def default_value(self):
+        return []
+
+    def get_validators(self, existing_registration):
+        def _check_number_of_sessions(new_data):
+            _min = self.view_data.get('minimum')
+            _max = self.view_data.get('maximum')
+            if _max is not None and _min is not None:
+                if _max != 0 and len(new_data) > _max:
```

Comment:
Since it's a numbered field in the settings form and we need a way to determine no-restriction, to let 0 as no restriction is the best approach I found.

it's not clear to me what you mean with `load_default=None`, yes, I can load None as default value, but then, if the user sets up a restriction, can they return it to no restriction in the settings form?

Reply:
After checking again, using 0 zero to mean no maximum is fine, we do the same thing for the text field. What would be nice though is showing a different text for 0 in the session field settings e.g. `No maximum` like we show in the text field settings:

![image](https://github.com/indico/indico/assets/8739637/5349d174-9789-4420-8bb5-553f53ea85af)

You can find the relevant logic in the `TextSettings` component :)